src/e2ee.rs

```diff
@@ -59,289 +60,248 @@ impl E2eeHelper {
-                                        encrypted_part;
-                                    (*encrypted_part).mm_parent = in_out_message;
-                                    mailmime_free(message_to_encrypt);
-                                    self.encryption_successfull = true;
-                                }
                             }
                         }
-                        if ok_to_continue {
-                            let aheader = Aheader::new(addr, public_key, prefer_encrypt);
-                            mailimf_fields_add(
-                                imffields_unprotected,
-                                mailimf_field_new_custom(
-                                    "Autocrypt".strdup(),
-                                    aheader.to_string().strdup(),
-                                ),
-                            );
+                        if move_to_encrypted {
+                            mailimf_fields_add(imffields_encrypted, field);
+                            cur = clist_delete((*imffields_unprotected).fld_list, cur);
+                        } else {
+                            cur = (*cur).next;
                         }
                     }
+                    let subject: *mut mailimf_subject = mailimf_subject_new("...".strdup());
+                    mailimf_fields_add(
+                        imffields_unprotected,
+                        mailimf_field_new(
+                            MAILIMF_FIELD_SUBJECT as libc::c_int,
+                            ptr::null_mut(),
+                            ptr::null_mut(),
+                            ptr::null_mut(),
+                            ptr::null_mut(),
+                            ptr::null_mut(),
+                            ptr::null_mut(),
+                            ptr::null_mut(),
+                            ptr::null_mut(),
+                            ptr::null_mut(),
+                            ptr::null_mut(),
+                            ptr::null_mut(),
+                            ptr::null_mut(),
+                            ptr::null_mut(),
+                            ptr::null_mut(),
+                            ptr::null_mut(),
+                            ptr::null_mut(),
+                            ptr::null_mut(),
+                            ptr::null_mut(),
+                            subject,
+                            ptr::null_mut(),
+                            ptr::null_mut(),
+                            ptr::null_mut(),
+                        ),
+                    );
+                    wrapmime::append_ct_param(
+                        (*part_to_encrypt).mm_content_type,
+                        "protected-headers",
+                        "v1",
+                    )?;
+                    mailmime_write_mem(plain, &mut col, message_to_encrypt);
+                    if (*plain).str_0.is_null() || (*plain).len <= 0 {
+                        bail!("could not write/allocate");
```

Comment:
this should panic, if we can't allocate memory we are out of options

Reply:
hum, thought so too but message_to_encrypt might simply be too big but otherwise the program can continue?

Reply:
fair enough, sth to look at in mmime directly